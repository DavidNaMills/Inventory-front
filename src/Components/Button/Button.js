import React from 'react';
import classes from './button.module.scss';

const allTypes = {
    base: {
        base: classes.button__base,
        shadow: classes.button__base_shadow,
        border: classes.button__base_border,
        skeleton: classes.button__base_skeleton,
    },

    submit: {
        base: classes.button__submit,
        shadow: classes.button__submit_shadow,
        border: classes.button__submit_border,
        skeleton: classes.button__submit_skeleton,
    },

    clear: {
        base: classes.button__clear,
        shadow: classes.button__clear_shadow,
        border: classes.button__clear_border,
        skeleton: classes.button__clear_skeleton,
    },

    danger: {
        base: classes.button__danger,
        shadow: classes.button__danger_shadow,
        border: classes.button__danger_border,
        skeleton: classes.button__danger_skeleton,
    },

    information: {
        base: classes.button__information,
        shadow: classes.button__information_shadow,
        border: classes.button__information_border,
        skeleton: classes.button__information_skeleton,
    },
};


const Button = ({ config = {}, onClick, label = null, skeleton = false, raised = false, full = false, type = 'base', border = false, bold = false, wide = false, isDisabled = false, Icon = null, noStyle=null }) => {
    const fullStyle = [classes.button];
    let which = type;
    const style = {};

    if (isDisabled) {
        fullStyle.push(classes.button__disabled);
    } else if(noStyle) { fullStyle.push(classes.button_noStyle);
    } else {
        if (allTypes[type]) {
            which = type;
            fullStyle.push(allTypes[type].base);
        } else {
            which = 'base';
            fullStyle.push(allTypes['base'].base);
        }

        if (skeleton) { fullStyle.push(allTypes[which].skeleton); }
        if (raised) { fullStyle.push(allTypes[which].shadow); }
        if (border) { fullStyle.push(allTypes[which].border); }
    }
    if (full) { fullStyle.push(classes.button_full); }
    if (bold) { fullStyle.push(classes.button_bold); }
    
    
    if (wide) {
        style.width = '30%';
    }

    return (
        <button
            {...config}
            onClick={onClick}
            style={style}
            className={fullStyle.join(' ')}
            disabled={isDisabled}
        ><p>{label}</p>{Icon && <Icon />}</button>
    )
}

export default React.memo(Button);