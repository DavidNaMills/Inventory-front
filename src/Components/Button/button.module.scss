@import '../../styles/mixins/buttonMixins.module.scss';
@import '../../styles/variables/colors.module.scss';

.button{
    display: flex;
    align-content: center;
    justify-content: center;
    font-size: 16px;
    border: none;
    padding: 10px;
    padding-left: 20px;
    padding-right: 20px;
    border-radius: 2px;

    p{
        padding-top: 5px;
        padding-right: 10px;
    }

    &__disabled{
        background-color: $disabled-back;
        color: $disabled-color;
    }

    &__base{
        @include buttonStyle($base-back, $base-color);
        &:hover{
            @include onHover($base-shadow, $light-text, $base-back);
        }

        &_shadow{@include buttonShadow($base-shadow);}
        &_border{@include buildBorder($base-color);}
        &_skeleton{@include buildSkeleton($base-back, $base-color); }
    };
    
    &__submit{
        @include buttonStyle($submit-back, $submit-color);
        &:hover{
            @include onHover($submit-shadow, $light-text, $submit-back);
        }

        &_shadow{ @include buttonShadow($submit-shadow); }
        &_border{ @include buildBorder($submit-color); }
        &_skeleton{@include buildSkeleton($submit-back, $submit-color); }
    }
    
    &__clear{
        @include buttonStyle($clear-back, $clear-color);
        &:hover{
            @include onHover($clear-shadow, $clear-color, $clear-back);
        }

        &_shadow{ @include buttonShadow($clear-shadow); }
        &_border{ @include buildBorder($clear-color); }
        &_skeleton{@include buildSkeleton($clear-back, $clear-color); }
    }
    
    &__danger{
        @include buttonStyle($danger-back, $danger-color);
        &:hover{
            @include onHover($danger-shadow, $light-text, $danger-back);
        }
        &_shadow{ @include buttonShadow($danger-shadow); }
        &_border{ @include buildBorder($danger-color); }
        &_skeleton{@include buildSkeleton($danger-back, $danger-color); }
    }
    
    &__information{
        @include buttonStyle($information-back, $information-color);
        &:hover{
            @include onHover($information-shadow, $light-text, $information-back);
        }
        &_shadow{ @include buttonShadow($information-shadow); }
        &_border{ @include buildBorder($information-color); }
        &_skeleton{@include buildSkeleton($information-back, $information-color); }
    }

    &_flat{
        border: none;
        box-shadow: none;
        -webkit-box-shadow: none;
        -moz-box-shadow: none;
    }

    &_full{
        width: 100%;
    }

    &_bold{
        font-weight: bold;
    }

    &_noStyle{
        background-color: transparent;
        border: none;
        box-shadow: none;

        &:hover{
            background-color: rgb(212, 212, 212);
            padding-top: 10px;
            padding-bottom: 10px;
            padding-left: 5px;
            padding-right: 10px;
            border-radius: 100px;
        }
    }
}