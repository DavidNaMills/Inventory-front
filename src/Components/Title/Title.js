import React from 'react';
import classes from './title.module.scss';


const Title = (props) => {
    const { size = null, message = '', center = false } = props;
    let element = null;

    switch (size) {
        case 'h1':
            element = <h1 className={[classes.title_h1, center ? classes.title__center : null].join(' ')} >{message}</h1>
            break;
        case 'h2':
            element = <h2 className={[classes.title_h2, center ? classes.title__center : null].join(' ')}>{message}</h2>
            break;
        case 'h3':
            element = <h3 className={[classes.title_h3, center ? classes.title__center : null].join(' ')}>{message}</h3>
            break;
        case 'h4':
            element = <h4 className={[classes.title_h4, center ? classes.title__center : null].join(' ')}>{message}</h4>
            break;
        case 'h5':
            element = <h5 className={[classes.title_h5, center ? classes.title__center : null].join(' ')}>{message}</h5>
            break;
        case 'h6':
            element = <h6 className={[classes.title_h6, center ? classes.title__center : null].join(' ')}>{message}</h6>
            break;
        default:
            element = <h2 className={[classes.title_h2, center ? classes.title__center : null].join(' ')}>{message}</h2>
            break;
    };

    return (
        <div>
            {element}
        </div>
    )
}

export default React.memo(Title);