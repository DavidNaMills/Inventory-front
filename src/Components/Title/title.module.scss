@import '../../styles/variables/colors.module.scss';
@import '../../styles/fonts/fontSizes.module.scss';
@import '../../styles/mixins/title.module.scss';

.title {
    @include setTitle($h2, $title-color);
    
    &_h1{
        @include setTitle($h1, $title-color);
    };

    &_h2{
        @include setTitle($h2, $title-color);
    };
    
    &_h3{
        @include setTitle($h3, $title-color);
    }

    &_h4{
        @include setTitle($h4, $title-color);
    }

    &_h5{
        @include setTitle($h5, $title-color);
    }

    &_h6{
        @include setTitle($h6, $title-color);
    }

    &__center{
        text-align: center;
    }


}