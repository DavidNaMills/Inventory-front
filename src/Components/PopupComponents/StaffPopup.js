import React from 'react';
import classes from './Popup.module.scss';
import { useSelector } from 'react-redux';
import Title from '../Title/Title';
import Button from '../Button/Button';
import { useLocale } from '../../Context/LocaleContext';


const tempStyle = {
    backgroundColor: 'rgb(237, 237, 237)',
    padding: '50px',
    border: 'solid black 2px',
    minWidth: '80%'
}

const bold = {
    fontWeight: 'bold'
}

const locs = {
    border: 'solid black 1px',
    backgroundColor: 'white',
    padding: '10px'
}

const StaffPopup = (props) => {
    const { item, onClose } = props;
    const { types } = useSelector(state => state);
    const locale = useLocale();

    const isRole = (key, txt) =>{
        if(key === 'role'){
            return locale[txt];
        } else {
            return txt;
        }
    }

    return (
        // <div style={tempStyle}>
        <div className={classes.popup}>
            <Title message={item.name} size='h2' center />
            <div className={classes.popup__smallSpacing}>
                <label style={bold}>{locale['role']}</label>
                <p>{locale[item.role]}</p>
            </div>
            <div className={classes.popup__smallSpacing}>
                <label style={bold}>{locale['phone']}</label>
                <p>{item.phone}</p>
            </div>
            <div className={classes.popup__mediumSpacing}>
                <label style={bold}>{locale['frm_staff_wechat']}</label>
                <p>{item.wechat}</p>
            </div>

            <div style={locs} className={classes.popup__mediumSpacing}>
            <label style={bold}>{locale['location']}</label>
                {
                    item.baseId.map(x => <p key={x}>{types.locations[x].displayValue}</p>)
                }
            </div>
            <Button onClick={onClose} label={locale['close']} full />
        </div>
    )
}

export default StaffPopup;