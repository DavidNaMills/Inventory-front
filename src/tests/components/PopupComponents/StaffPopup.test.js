import React from 'react';
import '../../testHelpers/context';
import '../../Containers/containerHelper';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

import StaffPopup from '../../../Components/PopupComponents/StaffPopup';
import Title from '../../../Components/Title/Title';
import Button from '../../../Components/Button/Button';


const staffMember = {
    name: 'david',
    role: 3,
    phone: '12312312312',
    wechat: 'g4fd56s78fg9ds',
    baseId: ['123', '456123']
};

const mockClose = jest.fn();

describe('<StaffPopup/> component test suite', ()=>{
    let wrapper;
    
    beforeEach(()=>{
        wrapper = shallow(<StaffPopup item={staffMember} onClose={mockClose}/>);
    });

    it('matches snapshot', ()=>{
        const comp = renderer.create(<StaffPopup item={staffMember} onClose={mockClose}/>);
        const app = comp.toJSON();
        expect(app).toMatchSnapshot();
    });
    
    it('exists', ()=>{
        expect(wrapper.exists).toBeTruthy();
    });
    
    it('renders a <div> with inline styles', ()=>{
        const div = wrapper.find('div');
        expect(div.length).toBe(5);
        expect(div.at(0).props().style).toEqual({
            backgroundColor: 'rgb(237, 237, 237)',
            padding: '50px',
            border: 'solid black 2px',
            minWidth: '80%'
        });
    });
    
    it('renders a <Title/> component', ()=>{
        const title = wrapper.find(Title);
        expect(title.length).toBe(1);
        expect(title.props()).toHaveProperty('message', staffMember.name);
        expect(title.props()).toHaveProperty('size', 'h2');
    });
    
    it('renders a <div> with classname "popup__smallSpacing" label and <p> for Role ', ()=>{
        const div = wrapper.find('div');
        expect(div.at(1).props()).toHaveProperty('className', 'popup__smallSpacing');
        expect(div.at(1).find('label').props().style).toHaveProperty('fontWeight', 'bold');
        expect(div.at(1).find('label').text()).toEqual('Role');
        expect(div.at(1).find('p').text()).toEqual(`${staffMember.role}`);

    });
    
    it('renders a <div> with classname "popup__smallSpacing" label and <p> for Phone ', ()=>{
        const div = wrapper.find('div');
        expect(div.at(2).props()).toHaveProperty('className', 'popup__smallSpacing');
        expect(div.at(2).find('label').props().style).toHaveProperty('fontWeight', 'bold');
        expect(div.at(2).find('label').text()).toEqual('Phone');
        expect(div.at(2).find('p').text()).toEqual(`${staffMember.phone}`);
    });
    
    it('renders a <div> with classname "popup__mediumSpacing" label and <p> for wechat ', ()=>{
        const div = wrapper.find('div');
        expect(div.at(3).props()).toHaveProperty('className', 'popup__mediumSpacing');
        expect(div.at(3).find('label').props().style).toHaveProperty('fontWeight', 'bold');
        expect(div.at(3).find('label').text()).toEqual('Wechat');
        expect(div.at(3).find('p').text()).toEqual(`${staffMember.wechat}`);
    });
    
    it('renders a <div> with classname "popup__mediumSpacing" label and 2 <p> for locations ', ()=>{
        const div = wrapper.find('div');
        expect(div.at(4).props()).toHaveProperty('className', 'popup__mediumSpacing');
        expect(div.at(4).find('label').props().style).toHaveProperty('fontWeight', 'bold');
        expect(div.at(4).find('label').text()).toEqual('Locations');
        expect(div.at(4).find('p').length).toBe(2);
    });
    
    it('renders a <Button /> component', ()=>{
        const btn = wrapper.find(Button);
        expect(btn.length).toBe(1);
        expect(btn.props()).toHaveProperty('label', 'Close');
        expect(btn.props()).toHaveProperty('full', true);
        expect(btn.props()).toHaveProperty('onClick', mockClose);
    });
});