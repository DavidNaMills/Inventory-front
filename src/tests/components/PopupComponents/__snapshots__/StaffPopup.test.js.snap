// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StaffPopup/> component test suite matches snapshot 1`] = `
<div
  style={
    Object {
      "backgroundColor": "rgb(237, 237, 237)",
      "border": "solid black 2px",
      "minWidth": "80%",
      "padding": "50px",
    }
  }
>
  <div>
    <h2
      className="title_h2 title__center"
    >
      david
    </h2>
  </div>
  <div
    className="popup__smallSpacing"
  >
    <label
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      role
    </label>
    <p>
      Admin
    </p>
  </div>
  <div
    className="popup__smallSpacing"
  >
    <label
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      phone
    </label>
    <p />
  </div>
  <div
    className="popup__mediumSpacing"
  >
    <label
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Wechat
    </label>
    <p />
  </div>
  <div
    className="popup__mediumSpacing"
    style={
      Object {
        "backgroundColor": "white",
        "border": "solid black 1px",
        "padding": "10px",
      }
    }
  >
    <label
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Locations
    </label>
    <p>
      home
    </p>
    <p>
      home2
    </p>
  </div>
  <button
    className="button button__base button_full"
    disabled={false}
    onClick={[MockFunction]}
    style={Object {}}
  >
    <p>
      Close
    </p>
  </button>
</div>
`;
